pkg/scheduler/eventhandlers.go

```diff
@@ -246,16 +249,39 @@ func responsibleForPod(pod *v1.Pod, profiles profile.Map) bool {
 	return profiles.HandlesSchedulerName(pod.Spec.SchedulerName)
 }
 
+const (
+	// syncedPollPeriod controls how often you look at the status of your sync funcs
+	syncedPollPeriod = 100 * time.Millisecond
+)
+
+// WaitForHandlersSync waits for EventHandlers to sync.
+// It returns true if it was successful, false if the controller should shut down
+func (sched *Scheduler) WaitForHandlersSync(ctx context.Context) error {
+	return wait.PollUntilContextCancel(ctx, syncedPollPeriod, true, func(ctx context.Context) (done bool, err error) {
+		for _, handler := range sched.registeredHandlers {
+			if !handler.HasSynced() {
```

Comment:
Can we have some v(3) that tell us which handlers are not synced? Ideally not for every iteration, but at least once per second.

Reply:
Current `registration` doesn't export a `Name` to us.
```
type ResourceEventHandlerRegistration interface {
	// HasSynced reports if both the parent has synced and all pre-sync
	// events have been delivered.
	HasSynced() bool
}

```
So in order to tell which handler is not synced, we need to create a new struct
```
type namedRegistration struct {
	cache.ResourceEventHandlerRegistration
	Name string
}
```
And create a name for every handler like:
```
handlers = append(handlers, namedRegistration{
	ResourceEventHandlerRegistration: handlerRegistration,
	Name:                             "dynInformer",
})
```
Is this way ok?